crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -770,36 +770,23 @@ impl<'src> Parser<'src> {
     ///
     /// If the parser isn't positioned at a `type` token.
     ///
-    /// See: <https://docs.python.org/3/reference/simple_stmts.html#grammar-token-python-grammar-type_stmt>
+    /// See: <https://docs.python.org/3/reference/simple_stmts.html#the-type-statement>
     fn parse_type_alias_statement(&mut self) -> ast::StmtTypeAlias {
         let start = self.node_start();
         self.bump(TokenKind::Type);
 
-        let (tok, tok_range) = self.next_token();
-        let name = if let Tok::Name { name } = tok {
-            Expr::Name(ast::ExprName {
-                id: name.to_string(),
-                ctx: ExprContext::Store,
-                range: tok_range,
-            })
-        } else {
-            // TODO(dhruvmanila): This recovery isn't possible currently because the soft keyword
-            // transformer will always convert the `type` token to a `Name` token if it's not
-            // followed by a `Name` token.
-            self.add_error(
-                ParseErrorType::OtherError(format!("expecting identifier, got {tok}")),
-                tok_range,
-            );
-            Expr::Name(ast::ExprName {
-                id: String::new(),
-                ctx: ExprContext::Invalid,
-                range: tok_range,
-            })
-        };
+        let mut name = Expr::Name(self.parse_name());
+        helpers::set_expr_ctx(&mut name, ExprContext::Store);
+
         let type_params = self.try_parse_type_params();
 
         self.expect(TokenKind::Equal);
 
+        // TODO(dhruvmanila): Disallow yield and starred expression
+        // test_err type_alias_invalid_value_expr
+        // type x = *y
+        // type x = yield y
+        // type x = x := 1
         let value = self.parse_conditional_expression_or_higher();
 
         ast::StmtTypeAlias {
```

Comment:
Could we add some more invalid test cases:

```
type 
type x
type x = 
```

Reply:
The problem with some of these cases (the top two in your list) is that the `type` is not a keyword but a name expression because of the soft keywords transformer. So, the parser never sees this as a type alias statement (at least for the first two statement).

That said, I think it makes sense to still add them either with a comment or as is so that in the future we can improve this.